### PR TITLE
tests: modem: backend: Add fixture to test suite

### LINKED ISSUE #65559: subsys: modem: backends: uart: modem.backends.uart.isr and modem.backends.uart.async fail
**Describe the bug**
The tests `modem.backends.uart.isr` and `modem.backends.uart.async` fail.

Observed for `nrf5340dk_nrf5340_cpuapp`.

**To Reproduce**
1. have `nrf5340dk_nrf5340_cpuapp` connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/subsys/modem/backends/uart -p nrf5340dk_nrf5340_cpuapp --device-testing --device-serial /dev/ttyACM1 -v --inline-logs`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
*** Booting Zephyr OS build zephyr-v3.5.0-1863-g23324fc5bae4 ***
Running TESTSUITE modem_backend_uart_suite
===================================================================
START - test_transmit_receive
TX: 2,2
TX ACC: 2
```

**Environment (please complete the following information):**
- OS: Ubuntu 20.04.1 LTS
- Toolchain Zephyr SDK 0.16.1
- Commit SHA or Version used: zephyr-v3.5.0-1863-g23324fc5bae4

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
The tests in this testcase require physical modification of the boards to be performed. This is now documented using the `fixture: gpio_loopback` and `notes:` properties.

Fixes: #65559

### CODE REVIEW

**Comment by bjarki-trackunit** [comment]
@PerMac Very cool feature with the fixtures, however, I am unsure how to properly document the fixtures, specifically the `map.yml` file, the user has to modify it to fit their system, but I want to include the notes and general setup, how do I do that? I have simply copied the file as I used it for now :)

**Comment by PerMac** [comment]
There is no documentation for fixture. The problem is there is no unified definition of what specific keywords should mean. They are also not used consistently. As for the map.yaml: this is meant to be a local file which describes your setup. Those are not added to the tree since everyone will have an individual setup.

**Comment by PerMac** [comment]
I believe you can use a "gpio_loopback" for this (you put this in the pr description but used uart_loopback in the yaml). Could you please align pins in the overlay and the name of the fixture with https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/uart/uart_async_api/testcase.yaml? this way we will be able to run both of them on the same hw setup.

**Comment by bjarki-trackunit** [comment]
I have aligned the fixture to match the ones used for the uart_async_api tests, and removed the map.yml file :) The modem backend test is now compatible with those fixtures :)

**Comment by bjarki-trackunit** [comment]
Again, love this feature, not sure how its possible I have not come across it before :)

**Comment by bjarki-trackunit** [comment]
Output from running the tests for both fixtures :)

```
Renaming output directory to /home/private/Documents/trackunit/zephyr_rtc/zephyr/twister-out.49
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.5.0-2055-g44cb294de09d
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/private/Documents/trackunit/zephyr_rtc/zephyr/twister-out/testplan.json

Device testing on:

| Platform                 |   ID | Serial device   |
|--------------------------|------|-----------------|
| b_u585i_iot02a           |    1 | /dev/ttyACM3    |
| nrf5340dk_nrf5340_cpuapp |    2 | /dev/ttyACM1    |

INFO    - JOBS: 24
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    4/   4  100%  skipped:    0, failed:    0, error:    0
INFO    - 2 test scenarios (4 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 4 of 4 test configurations passed (100.00%), 0 failed, 0 errored, 0 skipped with 0 warnings in 35.99 seconds
INFO    - In total 4 test cases were executed, 0 skipped on 2 out of total 647 platforms (0.31%)
INFO    - 4 test configurations executed on platforms, 0 test configurations were only built.

Hardware distribution summary:

| Board                    |   ID |   Counter |
|--------------------------|------|-----------|
| b_u585i_iot02a           |    1 |         2 |
| nrf5340dk_nrf5340_cpuapp |    2 |         2 |
INFO    - Saving reports...
INFO    - Writing JSON report /home/private/Documents/trackunit/zephyr_rtc/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/private/Documents/trackunit/zephyr_rtc/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/private/Documents/trackunit/zephyr_rtc/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

**Review by PerMac — DISMISSED** [review]

File: tests/subsys/modem/backends/uart/testcase.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+    - nrf5340dk_nrf5340_cpuapp
+
 tests:
   modem.backends.uart.async:
-    tags: modem_backend
-    harness: ztest
-    platform_allow:
-      - b_u585i_iot02a
-      - nrf5340dk_nrf5340_cpuapp
+    tags: modem_backend_uart_async
```
Comment (PerMac) [inline tests/subsys/modem/backends/uart/testcase.yaml:14]: do you need this tag somewhere? the story with tags is a bit like with fixtures, there are no written rules for them. They can be used for filtering tests in twister. Will ones you propose bring any value? The name is quite lengthy and looks very specific. If it is the single place it exist it won't be that useful.

File: tests/subsys/modem/backends/uart/testcase.yaml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
     extra_configs:
       - CONFIG_UART_ASYNC_API=y
 
   modem.backends.uart.isr:
-    tags: modem_backend
-    harness: ztest
-    platform_allow:
-      - b_u585i_iot02a
-      - nrf5340dk_nrf5340_cpuapp
+    tags: modem_backend_uart_isr
```
Comment (PerMac) [inline tests/subsys/modem/backends/uart/testcase.yaml:19]: ditto

Comment (bjarki-trackunit) [reply tests/subsys/modem/backends/uart/testcase.yaml]: Removed

**Review by PerMac — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**